streams-bootstrap-test/src/main/java/com/bakdata/kafka/StreamsBootstrapTopologyFactory.java

```diff
@@ -0,0 +1,111 @@
+     */
+    public static <K, V> TestTopology<K, V> createTopologyWithSchemaRegistry(final KafkaStreamsApplication app) {
+        return new TestTopology<>(p -> createTopology(app, p), app.getKafkaProperties());
+    }
+
+    /**
+     * Create a {@code TestTopologyExtension} from a {@code KafkaStreamsApplication}. This also sets
+     * {@link KafkaStreamsApplication#schemaRegistryUrl}.
+     *
+     * @param app KafkaStreamsApplication to create TestTopology from
+     * @param <K> Default type of keys
+     * @param <V> Default type of values
+     * @return {@code TestTopologyExtension} that uses topology and configuration provided by
+     * {@code KafkaStreamsApplication}
+     * @see KafkaStreamsApplication#getKafkaProperties()
+     * @see KafkaStreamsApplication#createTopology()
+     */
+    public static <K, V> TestTopologyExtension<K, V> createTopologyExtensionWithSchemaRegistry(
+            final KafkaStreamsApplication app) {
+        //FIXME properties do not use correct default serde because SR url is not yet set
+        return new TestTopologyExtension<>(p -> createTopology(app, p), app.getKafkaProperties());
+    }
+
+    /**
+     * Create a {@code TestTopology} from a {@code KafkaStreamsApplication}. This does not set
+     * {@link KafkaStreamsApplication#schemaRegistryUrl}.
+     *
+     * @param app KafkaStreamsApplication to create TestTopology from
+     * @param <K> Default type of keys
+     * @param <V> Default type of values
+     * @return {@code TestTopology} that uses topology and configuration provided by {@code KafkaStreamsApplication}
+     *
+     * @see KafkaStreamsApplication#getKafkaProperties()
+     * @see KafkaStreamsApplication#createTopology()
+     */
+    public static <K, V> TestTopology<K, V> createTopology(final KafkaStreamsApplication app) {
+        return new TestTopology<>(app::createTopology, app.getKafkaProperties());
+    }
+
+    /**
+     * Create a {@code TestTopologyExtension} from a {@code KafkaStreamsApplication}. This does not set
+     * {@link KafkaStreamsApplication#schemaRegistryUrl}.
+     *
+     * @param app KafkaStreamsApplication to create TestTopology from
+     * @param <K> Default type of keys
+     * @param <V> Default type of values
+     * @return {@code TestTopologyExtension} that uses topology and configuration provided by {@code
+     * KafkaStreamsApplication}
+     *
+     * @see KafkaStreamsApplication#getKafkaProperties()
+     * @see KafkaStreamsApplication#createTopology()
+     */
+    public static <K, V> TestTopologyExtension<K, V> createTopologyExtension(final KafkaStreamsApplication app) {
+        return new TestTopologyExtension<>(app::createTopology, app.getKafkaProperties());
+    }
+
+    private static Topology createTopology(final KafkaStreamsApplication app, final Properties properties) {
+        app.setSchemaRegistryUrl(properties.getProperty(AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG));
```

Comment:
Can we use the [SR's mock protocol](https://github.com/confluentinc/schema-registry/blob/764e345f1f943065801d5034f285c3ad4156eaaf/client/src/main/java/io/confluent/kafka/schemaregistry/testutil/MockSchemaRegistry.java#L39) as Torben suggested?

Reply:
Something like this:
```suggestion
        app.setSchemaRegistryUrl("mock://test-scope");
```

Reply:
No, this is about using the SR mock provided with fluent Kafka

Reply:
We should rather evaluate whether we can get rid of SR mock in fluent Kafka altogether